File: commands/get_status.go
// Copyright (c) 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package commands

import "cocoon/db"

// GetStatusCommand gets dashboard status.
type GetStatusCommand struct {
}

// GetStatusResult contains dashboard status.
type GetStatusResult struct {
	Statuses      []*db.BuildStatus
	AgentStatuses []*db.AgentStatus
}

// GetStatus returns current build status.
func GetStatus(c *db.Cocoon, _ []byte) (interface{}, error) {
	var err error

	statuses, err := c.QueryBuildStatusesWithMemcache()

	if err != nil {
		return nil, err
	}

	agentStatuses, err := c.QueryAgentStatuses()

	if err != nil {
		return nil, err
	}

	return &GetStatusResult{
		Statuses:      statuses,
		AgentStatuses: agentStatuses,
	}, nil
}

File: app_dart/lib/src/model/rpc/get_status.dart
// Copyright 2019 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:json_annotation/json_annotation.dart';

import '../appengine/agent.dart';
import '../appengine/commit_status.dart';

part 'get_status.g.dart';


@JsonSerializable()
class GetStatusResponse {
  GetStatusResponse({this.agents, this.statuses = const <CommitStatus>[]});

  factory GetStatusResponse.fromJson(Map<String, dynamic> json) => _$GetStatusResponseFromJson(json);

  final List<Agent> agents;
  final List<CommitStatus> statuses;

  Map<String, dynamic> toJson() => _$GetStatusResponseToJson(this);
}